httpbp/headers.go

```diff
@@ -0,0 +1,188 @@
+package httpbp
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	edgeContextHTTPHeaderKey = "X-Edge-Request"
+	parentIDHTTPHeaderKey    = "X-Parent"
+	spanIDHTTPHeaderKey      = "X-Span"
+	spanFlagsHTTPHeaderKey   = "X-Flags"
+	spanSampledHTTPHeaderKey = "X-Sampled"
+	traceIDHTTPHeaderKey     = "X-Trace"
+)
+
+type headerContextKey int
+
+const (
+	edgeContextContextKey headerContextKey = iota
+	traceIDContextKey
+	parentIDContextKey
+	spanIDContextKey
+	spanFlagsContextKey
+	spanSampledContextKey
+)
+
+// HTTPHeaderKey is an Enum used to get HTTP headers from a context object.
+//
+// It is not used as the actual key in the context, these enums are mapped to
+// un-exported values that serve as the actual context keys.
+type HTTPHeaderKey int
+
+const (
+	// EdgeContextHTTPHeader is the key for the raw edge request context
+	EdgeContextHTTPHeader HTTPHeaderKey = iota
+
+	// TraceIDHTTPHeader is the header for the trace ID passed by the caller
+	TraceIDHTTPHeader
```

Comment:
these two are with suffix `HTTPHeader` whole others are only with suffix `Header`. I think since this is `httpbp` package we could just use `Header` suffix consistently.